adoc-mode.el

```diff
@@ -2018,14 +2032,14 @@ START-SRC and END-SRC delimit the actual source code."
             (insert string))
           (unless (eq major-mode lang-mode) (funcall lang-mode))
           (font-lock-ensure)
-          (cl-loop for int being the intervals property 'face
-                   for pos = (car int)
-                   for next = (cdr int)
-                   for val = (get-text-property pos 'face)
-                   when val do
-                   (put-text-property
-                    (+ start-src (1- pos)) (1- (+ start-src next)) 'face
-                    val adoc-buffer)))
+          (adoc-map-intervals
+           (lambda (pos next)
+             (let ((val (get-text-property pos 'face)))
```

Comment:
There's also `when-let` that would simplify this code a bit.

Reply:
I know `when-let` from but I didn't know whether you like it.
It was long in `subr-x.el` instead of `subr.el`. Therefore, it was considered as seldom used feature:

> commit b05a103ea7a26b2f4099a613015d9f1abdc39a4d
Author: Lars Ingebrigtsen <larsi@gnus.org>
Date:   Sat Apr 30 16:42:44 2022 +0200
>
>    Move the when-let family of macros to subr.el
>    
>    * lisp/subr.el (internal--build-binding)
    (internal--build-bindings): Moved from subr-x.el and rewritten to
    not use the threading macro.
    (if-let*, when-let*, and-let*, if-let, when-let): Moved from
    subr-x.el.  This avoids breaking the build every time somebody
    uses these macros in functions that end up being called during
    bootstrap.